cli/internal/env/env.go

```diff
@@ -31,9 +31,13 @@ func getEnvPairsFromKeys(envKeys []string, allEnvVars map[string]string) []strin
 }
 
 // getFrameworkEnvPairs returns a slice of all key=value pairs that match the given prefix
-func getEnvPairsFromPrefix(prefix string, allEnvVars map[string]string) []string {
+func getEnvPairsFromPrefix(prefix string, allEnvVars map[string]string, blockVendorKey bool) []string {
```

Comment:
I don't love `boolean` arguments that change function behavior, it's basically two separate but closely-related functions in that world. (This is why I pitched the filter as a separate pass.)

Can we make this subordinate function receive all of its arguments rather than going to look it up in `allEnvVars`? (e.g. this function depends on X, Y, and Z inputs, it just so happens that Z is a subkey of Y, but that's incidental, and would break in a react-like call if identity of Y didn't change.)

Actual concrete suggestion: how about passing in `includePrefix` and `excludePrefix`?

Reply:
👍🏼 I like this. good suggestion.